deployment/address_book.go

```diff
@@ -93,20 +93,22 @@ type AddressBookMap struct {
 	mtx              sync.RWMutex
 }
 
-// save will save an address for a given chain selector. It will error if there is a conflicting existing address.
-func (m *AddressBookMap) save(chainSelector uint64, address string, typeAndVersion TypeAndVersion) error {
-	_, exists := chainsel.ChainBySelector(chainSelector)
-	if !exists {
+// Save will save an address for a given chain selector. It will error if there is a conflicting existing address.
+func (m *AddressBookMap) Save(chainSelector uint64, address string, typeAndVersion TypeAndVersion) error {
+	family, err := chainsel.GetSelectorFamily(chainSelector)
+	if err != nil {
 		return errors.Wrapf(ErrInvalidChainSelector, "chain selector %d", chainSelector)
 	}
-	if address == "" || address == common.HexToAddress("0x0").Hex() {
-		return errors.Wrap(ErrInvalidAddress, "address cannot be empty")
-	}
-	if common.IsHexAddress(address) {
-		// IMPORTANT: WE ALWAYS STANDARDIZE ETHEREUM ADDRESS STRINGS TO EIP55
-		address = common.HexToAddress(address).Hex()
-	} else {
-		return errors.Wrapf(ErrInvalidAddress, "address %s is not a valid Ethereum address, only Ethereum addresses supported", address)
+	if family == chainsel.FamilyEVM {
```

Comment:
could add your aptos specific validation?

Reply:
I was toying with an idea of building in aptos and other chain or family validations into the chainlink-selectors repo, so we'd have a set of reusable validations keyed off of chain ID in a way that hides all the conditionality from calling code. Valid addresses was definitely in the right zone.  Doing it here is fine, but we should think about making this more generally usable.